components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -5,6 +5,61 @@ use crate::codec::{self, Error};
 use crate::expr::EvalContext;
 use crate::Result;
 
+#[inline]
+#[rpn_fn]
+pub fn log_1_arg(arg: &Option<Real>) -> Result<Option<Real>> {
+    Ok(arg.and_then(|n| {
+        let res = n.ln();
+        if res.is_finite() {
```

Comment:
Seems `if res.is_finite()` is repeated lots of times, would you mind extract this logic?